### ISSUE TITLE
[bug] Version components prefixed with zeros are not considered valid anymore

### ISSUE THREAD

**Issue by fivaldi** [Masterminds/semver#255]
Accto https://helm.sh/docs/chart_template_guide/function_list/#semvercompare , helm uses this (Masterminds/semver) library. 

~~After switching helm from **3.15.3** to **3.16.4**, the following issue occurred:~~
After switching helmfile from **0.169.1** to **0.169.2**, the following issue occurred:

E.g. `01.2.3` or `1.02.3` or `1.2.03` are not valid anymore, though they're not against semver.  See the regex reference in https://semver.org/ and the linked examples: https://regex101.com/r/Ly7O1x/3/

![image](https://github.com/user-attachments/assets/afa9b073-a2d6-4274-acc7-6f84d0461a1d)

The regression appears after upgrading Masterminds/semver from **3.3.0** to **3.3.1**.

**Comment by yxxhero** [comment]
@mattfarina could you take a look? thanks so much.

**Comment by stevekuznetsov** [comment]
Seems like this was the commit that introduced the behavior: https://github.com/Masterminds/semver/commit/252dd61dd30c2bcf2bcfc30be1428c68d8f25934

**Comment by prathikshetty9b** [comment]
According to the Semantic Versioning (Semver) specification, version strings should not contain leading zeroes. Refer to SemVer specification section https://semver.org/#spec-item-2

**Comment by yxxhero** [comment]
@fivaldi https://semver.org/#spec-item-2

**Comment by fivaldi** [comment]
My bad, scrolling too fast over-seeing the "Invalid Semantic Versions" header. :( Sorry for this..

**Comment by haitch** [comment]
this is ubuntu versioning: 
![Image](https://github.com/user-attachments/assets/dd294db1-1851-476b-9d73-816686fb5fc7)

**Comment by fivaldi** [comment]
@haitch Good point. I've reopened the helm and helmfile issues (linked in the history of this issue), because I think the usage of [CalVer](https://calver.org/) in helm chart versions is legitimit.

I keep this one closed as this library validates strictly against semver.
